### PR TITLE
Http connect and read timeout

### PR DESCRIPTION
**Pull request by janakmulani**

### CODE REVIEW

File: core/src/main/java/net/sourceforge/jnlp/config/Defaults.java [hunk 1]
```diff
@@ -565,6 +565,18 @@
                     ConfigurationConstants.KEY_JVM_ARGS_WHITELIST,
                     null,
                     null
+            ),
+
+            Setting.createDefault(
+                    ConfigurationConstants.KEY_HTTPCONNECTION_CONNECT_TIMEOUT,
+                    String.valueOf(10000),
+                    null
```
Comment (sclassen) [inline core/src/main/java/net/sourceforge/jnlp/config/Defaults.java:573]: Can you use `ValidatorFactory.createRangedIntegerValidator(0, Integer.MAX_VALUE)`

File: core/src/main/java/net/sourceforge/jnlp/config/Defaults.java [hunk 2]
```diff
@@ -565,6 +565,18 @@
...
+            ),
+
+            Setting.createDefault(
+                    ConfigurationConstants.KEY_HTTPCONNECTION_READ_TIMEOUT,
+                    String.valueOf(10000),
+                    null
```
Comment (sclassen) [inline core/src/main/java/net/sourceforge/jnlp/config/Defaults.java:579]: same validator as for CONNECT_TIMEOUT

File: core/src/main/java/net/adoptopenjdk/icedteaweb/resources/downloader/BaseResourceDownloader.java [hunk 3]
```diff
@@ -152,7 +154,20 @@ private Resource tryDownloading(final URL downloadFrom) throws IOException {
...
+    }
+
+    private int getTimeoutValue(final String key) {
+        int timeout = 0;
+        final String value = JNLPRuntime.getConfiguration().getProperty(key);
+        if (value != null && value.trim().length() != 0) {
+            try {
+                timeout = Integer.valueOf(value);
+            } catch (NumberFormatException e) {
+                LOG.error(IcedTeaWebConstants.DEFAULT_ERROR_MESSAGE, e);
```
Comment (sclassen) [inline core/src/main/java/net/adoptopenjdk/icedteaweb/resources/downloader/BaseResourceDownloader.java:167]: Please use something else than `DEFAULT_ERROR_MESSAGE`, maybe
`Log.error("Could not pares {} with value {} - reason {}", key, value, e.getMessage());`

**Review by sclassen — CHANGES_REQUESTED** [review]

File: core/src/main/java/net/sourceforge/jnlp/config/Defaults.java (hunk 1 above)
Comment (janakmulani) [reply core/src/main/java/net/sourceforge/jnlp/config/Defaults.java]: done

File: core/src/main/java/net/sourceforge/jnlp/config/Defaults.java (hunk 2 above)
Comment (janakmulani) [reply core/src/main/java/net/sourceforge/jnlp/config/Defaults.java]: done

File: core/src/main/java/net/adoptopenjdk/icedteaweb/resources/downloader/BaseResourceDownloader.java (hunk 3 above)
Comment (janakmulani) [reply core/src/main/java/net/adoptopenjdk/icedteaweb/resources/downloader/BaseResourceDownloader.java]: done

**Pull request merged**